File: 1476.py
E, S, M = map(int,input().split())

e = 1
s = 1
m = 1 
year = 1

while True:
	if E == e and S == s and M == m:
		break
	
	year += 1
	curE = (e + 1) % 15
	curS = (s + 1) % 28
	curM = (m + 1) % 19
	
	if curE == 0:
		e = 15
	else:
		e = curE

	if curS == 0:
		s = 28
	else:
		s = curS
		
	if curM == 0:
		m = 19
	else:
		m = curM

print(year)

File: 1476.js
const stdin = `15 28 19`;
const [E, S, M] = stdin.split(' ').map(Number);

let e = 1;
let s = 1;
let m = 1;
let year = 1;

while (true) {
  if (E === e && S === s && M === m) {
    break;
  }
  year += 1;
  const curE = (e + 1) % 15;
  const curS = (s + 1) % 28;
  const curM = (m + 1) % 19;
  e = curE === 0 ? 15 : curE;
  s = curS === 0 ? 28 : curS;
  m = curM === 0 ? 19 : curM;
}

console.log(year);
